com.microsoft.java.debug.plugin/src/main/java/com/microsoft/java/debug/plugin/internal/ResolveMainMethodHandler.java

```diff
@@ -43,6 +47,17 @@ public static Object resolveMainMethods(List<Object> arguments) throws DebugExce
             return Collections.emptyList();
         }
 
+        // When the current document is changed, the language server will receive a didChange request about the changed text and then
+        // trigger a background job to update the change to the CompilationUnit. Because of race condition, the resolveMainMethods may read
+        // an old CompilationUnit. So add some waiting logic to wait the Document Update to finish first.
+        try {
+            Job.getJobManager().join(DocumentLifeCycleHandler.DOCUMENT_LIFE_CYCLE_JOBS, new NullProgressMonitor());
```

Comment:
Is there any chance to wait a smaller granularity of jobs of the current file?

Reply:
@yaohaizh is this possible?

Reply:
Not sure.

Reply:
Don't find the logic to wait on a smaller granularity job yet.